File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Client/src/models/userAccess/userAccess.ts
export class UserAccess {
    public id: string = crypto.randomUUID();
    public displayName: string = "";
    public entraGuid: string = "";
    public userPrincipalName: string = "";
    public entraUserId: string = "";
    public email: string = "";
    public givenName: string = "TODO";
    public surname: string = "TODO";
    public orgCodes: string[] = [];
    public orgCode: string = "";
    public jobTitle: string = "TODO";
    public activeFrom?: Date;
    public activeTo?: Date;
    public createdBy?: string;
    public createdDate?: Date | undefined;
    public updatedBy?: string;
    public updatedDate?: Date | undefined;
}

File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Server.Tests.Integration/Models/UserAccess/UserAccess.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

namespace ISL.ReIdentification.Configurations.Server.Tests.Integration.Models.UserAccess
{
    public class UserAccess
    {
        public Guid Id { get; set; }
        public string EntraUserId { get; set; }
        public string GivenName { get; set; } = string.Empty;
        public string Surname { get; set; } = string.Empty;
        public string DisplayName { get; set; } = string.Empty;
        public string JobTitle { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string UserPrincipalName { get; set; } = string.Empty;
        public string OrgCode { get; set; } = string.Empty;
        public string CreatedBy { get; set; } = string.Empty;
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
        public DateTimeOffset UpdatedDate { get; set; }
    }
}
